lib/vanagon/platform/windows.rb

```diff
@@ -53,14 +53,69 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Copy the project specific files first
+        wixworkdir = File.join(workdir, "wix")
+        copy_from_project("resources/windows/wix", wixworkdir, :verbose => true)
+
+        merge_defaults_from_vanagon("#{VANAGON_ROOT}/resources/windows/wix", wixworkdir, :verbose => true)
+        process_templates(wixworkdir, binding)
+      end
+
+      # Method to recursively copy from a source project resource directory
+      # to a destination (wix) work directory.
+      # strongly suspect the original cp_r command would have done all of this.
+      #
+      # @param proj_root [String] Root for the project resource files.
+      # @param destination [String] Destination directory
+      # @param verbose [String] True or false
+      def copy_from_project(proj_root, destination, verbose: false)
+        proj_path = Pathname.new(proj_root)
+        files = Dir.glob(File.join(proj_root, "**/*.*"))
+        files.each do |file|
+          src_pathname = Pathname.new(File.dirname(file))
+          dest_pathname = src_pathname.relative_path_from(proj_path)
+          target_dir = File.join(destination, dest_pathname.to_s)
+          FileUtils.mkdir_p(target_dir) unless File.exists?(target_dir)
+          FileUtils.cp(file, File.join(target_dir, File.basename(file)), :verbose => verbose)
+        end
```

Comment:
@jcoconnor if you think it's overcomplex for your needs, then rewrite it. I'd originally proposed a very small wrapper around `FileUtils.cp` for this function:

```ruby
def copy_from_project(*files, destination, verbose: false)
  FileUtils.cp_r(files, destination, :verbose => verbose)
end
```

And that could easily be rewritten to:

```

def copy_from_project(*files, destination, verbose: false)
  FileUtils.cp(files, destination, :verbose => verbose)
end
```

You don't need any of the sanity checking, normalization, path mangling, etc. inside of this function.